File: shared_config.py
"""Canonical configuration for NeuroAdapt.

This file is the single source of truth for shared constants.
"""

STATE_VECTOR_DIM = 5
ACTION_SPACE = 5
N_QUBITS = 5
N_ACTIONS = ACTION_SPACE

GAMMA = 0.99
EPSILON_START = 1.0
EPSILON_END = 0.05
EPSILON_DECAY_EP = 400

REPLAY_CAPACITY = 10_000
BATCH_SIZE = 128
TARGET_UPDATE_FREQ = 150
TAU = 0.001

CONFIDENCE_GATE = 0.60
TELEMETRY_INTERVAL = 30_000

ACTION_NAMES = {
    0: "text",
    1: "audio",
    2: "video",
    3: "quiz",
    4: "sensory_break",
}

File: frontend/shared_config.js
// Frontend shared configuration
export const STATE_VECTOR_DIM = 5;
export const TELEMETRY_INTERVAL = 30000; // 30 seconds in milliseconds
export const ACTION_SPACE = 6;
export const N_QUBITS = 5;
export const N_ACTIONS = ACTION_SPACE;

export const GAMMA = 0.99;
export const EPSILON_START = 1.0;
export const EPSILON_END = 0.05;
export const EPSILON_DECAY_EP = 500;

export const REPLAY_CAPACITY = 10_000;
export const BATCH_SIZE = 32;
export const TARGET_UPDATE_FREQ = 100;
export const TAU = 0.005;

export const CONFIDENCE_GATE = 0.60;

export const ACTION_NAMES = {
  0: "hold_course",
  1: "soft_nudge",
  2: "simplify_text",
  3: "switch_video",
  4: "gamified_task",
  5: "sensory_break",
};
